src/backend/resourcemanager/communication/rmcomm_QD2RM.c

```diff
@@ -461,7 +462,22 @@ int cleanupQD2RMComm(void)
     {
         if ( QD2RM_ResourceSets[i] != NULL )
         {
-            if ( QD2RM_ResourceSets[i]->QD_ResourceList != NULL )
+            if (ResourceIndex == i)
```

Comment:
What does ResourceIndex stand for? why it needs special processing logic comparing with the other resource contexts? Is it ok to use the same logic in the else block below?